packages/core/src/modules/connections/messages/ConnectionInvitationMessage.ts

```diff
@@ -61,12 +61,12 @@ export class ConnectionInvitationMessage extends AgentMessage {
     }
   }
 
-  @Equals(ConnectionInvitationMessage.type)
+  @IsValidMessageType(ConnectionInvitationMessage.type)
   @Transform(({ value }) => replaceLegacyDidSovPrefix(value), {
     toClassOnly: true,
   })
-  public readonly type = ConnectionInvitationMessage.type
-  public static readonly type = 'https://didcomm.org/connections/1.0/invitation'
+  public readonly type = ConnectionInvitationMessage.type.messageTypeUri
+  public static readonly type = parseMessageType('https://didcomm.org/connections/1.0/invitation')
```

Comment:
Not really part of your PR, but more just curious, why do we have a `public static readonly type` and `public readonly type`?

Reply:
Because we sometimes need the static property and sometimes the instance property. When transforming from/to a json object we need to set the `@type` property on the class.

However when we need to find the correct message class for an incoming message we don't have an instance yet, so we use the static  property. It's not ideal, but I'm not sure of other approaches. Any ideas? Would love to remove the need for both.

Reply:
I don't think I have another solution for this right now.